providers/go-feature-flag/pkg/provider.go

```diff
@@ -328,6 +331,9 @@ func evaluateWithRelayProxy[T model.JsonType](provider *Provider, ctx context.Co
 			// call to convertCache wouldn't result in the same error on the next call.
 			provider.cache.Remove(cacheKey)
 		} else {
+			if provider.cacheMetricsDisable {
+				return cacheValue
+			}
```

Comment:
The problem with this early return is that you are not changing the reason `cacheValue.Reason = of.CachedReason`. 

I think it may be better to put the if around the export.

Something like: 

```go
if !provider.cacheMetricsDisable {
 event := exporter.NewFeatureEvent(
				ffcontext.NewEvaluationContext(goffRequestBody.EvaluationContext.Key),
				flagName,
				cacheValue.Value,
				cacheValue.Variant,
				cacheValue.Reason == of.ErrorReason,
				"",
				"PROVIDER_CACHE",
 )
} ```